sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java

```diff
@@ -459,6 +476,7 @@ public StorageAccountImpl update() {
         updateParameters = new StorageAccountUpdateParameters();
         this.networkRulesHelper = new StorageNetworkRulesHelper(this.updateParameters, this.innerModel());
         this.encryptionHelper = new StorageEncryptionHelper(this.updateParameters, this.innerModel());
+        this.storageAccountMsiHandler = new StorageAccountMsiHandler(this.authorizationManager, this, createParameters, this.updateParameters, isInCreateMode());
```

Comment:
Why do we need a new MsiHandler instance in `update()`?

Reply:
Here you need to reference the `Storage Account` data in real time.

Reply:
Should you instead make `createParameters`, `updateParameters`, `isCreateMode()` in StorageAccountImpl package private, and do `StorageAccountMsiHandler(authorizationManager, this)`. Where in `StorageAccountMsiHandler` you call there variable/method in real time?

Reply:
The following adjustments have been made in the `StorageAccountMsiHandler` class:
1. The `isInCreateMode` fixed to get value via private method.
2. The construction method is simplified to `StorageAccountMsiHandler(final AuthorizationManager authorizationManager, StorageAccountImpl storageAccount)`
3. `updateParameters` and `createParameters` are changed to be set in the `withXXXXXXXX` methods, e.g. `withUpdateParameters`.